File: order/src/main/java/com/popcorn/order/config/SwaggerConfig.java
package com.popcorn.order.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import io.swagger.v3.oas.models.Components;
import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Contact;
import io.swagger.v3.oas.models.info.Info;
import io.swagger.v3.oas.models.info.License;
import io.swagger.v3.oas.models.security.SecurityRequirement;
import io.swagger.v3.oas.models.security.SecurityScheme;
import io.swagger.v3.oas.models.servers.Server;

/**
 * Order 마이크로서비스 Swagger/OpenAPI 설정
 */
@Configuration
public class SwaggerConfig {

    @Value("${spring.application.name}")
    private String applicationName;

    @Value("${app.gateway.url:${APP_GATEWAY_URL:https://api.goormpopcorn.shop}}")
    private String gatewayUrl;

    @Bean
    public OpenAPI openAPI() {
        return new OpenAPI()
                .info(apiInfo())
                .addServersItem(new Server()
                        .url(gatewayUrl)
                        .description("배포된 게이트웨이"))
                .components(new Components()
                        .addSecuritySchemes("bearer-token",
                                new SecurityScheme()
                                        .type(SecurityScheme.Type.HTTP)
                                        .scheme("bearer")
                                        .bearerFormat("JWT")
                                        .description("JWT 토큰을 입력하세요 (Bearer 접두사 제외)")))
                .addSecurityItem(new SecurityRequirement()
                        .addList("bearer-token"));
    }

    private Info apiInfo() {
        return new Info()
                .title("PopCorn Order Service API")
                .description("""
                        ## 🛒 PopCorn 주문 서비스 API

                        ### 📋 주요 기능
                        - **주문 생성**: 예약형(RESERVATION) / 구매형(GOODS) 주문 생성
                        - **주문 조회**: 사용자별, 상태별 주문 목록 및 상세 조회
                        - **주문 관리**: 상태 변경, 취소, 환불 처리
                        - **CQRS 패턴**: 명령과 조회 분리로 성능 최적화
                        - **이벤트 기반**: 주문 생성/변경 시 도메인 이벤트 발행

                        ### 🚀 API 사용 가이드
                        1. **주문 생성**: 팝업 예약 또는 굿즈 구매 주문 등록
                        2. **주문 조회**: 주문 ID나 사용자 ID로 주문 정보 확인
                        3. **주문 관리**: 주문 상태 변경, 취소 등 관리 기능

                        ### 🔒 인증 방식
                        - JWT Bearer Token 방식 사용
                        - Authorization 헤더에 "Bearer {token}" 형태로 전송

                        ### 📞 문의
                        - 개발팀: dev@popcorn.com
                        - 기술지원: support@popcorn.com
                        """)
                .version("v1.0.0")
                .contact(new Contact()
                        .name("PopCorn Development Team")
                        .email("dev@popcorn.com")
                        .url("https://github.com/popcorn-team"))
                .license(new License()
                        .name("MIT License")
                        .url("https://opensource.org/licenses/MIT"));
    }

}

File: payment/src/main/kotlin/com/popcorn/payment/config/SwaggerConfig.kt
package com.popcorn.payment.config

import io.swagger.v3.oas.models.Components
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import io.swagger.v3.oas.models.info.Contact
import io.swagger.v3.oas.models.info.License
import io.swagger.v3.oas.models.security.SecurityRequirement
import io.swagger.v3.oas.models.security.SecurityScheme
import io.swagger.v3.oas.models.servers.Server
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class SwaggerConfig {

    @Value("\${spring.application.name}")
    private lateinit var applicationName: String

    @Value("\${app.gateway.url:\${APP_GATEWAY_URL:https://api.goormpopcorn.shop}}")
    private lateinit var gatewayUrl: String

    @Bean
    fun openAPI(): OpenAPI {
        return OpenAPI()
            .info(apiInfo())
            .servers(listOf(Server().url(gatewayUrl).description("배포된 게이트웨이")))
            .components(
                Components()
                    .addSecuritySchemes("Bearer Authentication",
                        SecurityScheme()
                            .type(SecurityScheme.Type.HTTP)
                            .scheme("bearer")
                            .bearerFormat("JWT")
                            .description("JWT 토큰을 입력하세요")
                    )
            )
            .addSecurityItem(
                SecurityRequirement()
                    .addList("Bearer Authentication")
            )
    }

    private fun apiInfo(): Info {
        return Info()
            .title("PopCorn Payment Service API")
            .description("""
                ## 🎬 PopCorn 결제 서비스 API

                ### 📋 주요 기능
                - **토스페이먼츠 연동**: 카드, 간편결제, 가상계좌 등 다양한 결제 수단 지원
                - **멱등성 보장**: Redis 기반 중복 결제 방지
                - **비동기 처리**: Kotlin 코루틴 기반 성능 최적화
                - **장애 복구**: Circuit Breaker, Retry 패턴 적용
                - **실시간 모니터링**: 결제 상태 추적 및 알림

                ### 🚀 API 사용 가이드
                1. **결제 요청**: 주문 생성 후 결제 승인 호출
                2. **결제 상태 확인**: 주문 ID로 결제 진행 상황 조회
                3. **결제 취소**: 필요 시 전액 또는 부분 취소 처리

                ### 🔒 인증 방식
                - JWT Bearer Token 방식 사용
                - Authorization 헤더에 "Bearer {token}" 형태로 전송

                ### 📞 문의
                - 개발팀: dev@popcorn.com
                - 기술지원: support@popcorn.com
            """.trimIndent())
            .version("v1.0.0")
            .contact(
                Contact()
                    .name("PopCorn Development Team")
                    .email("dev@popcorn.com")
                    .url("https://github.com/popcorn-team")
            )
            .license(
                License()
                    .name("MIT License")
                    .url("https://opensource.org/licenses/MIT")
            )
    }
}
